### ISSUE TITLE
Install command to explicitly trigger a protoc install

### ISSUE THREAD

**Issue by lwc** [uber/prototool#202]
Hello, thanks for a great tool!

It's super handy being able to pin the version of `protoc` so easily, one thing we've noticed though is that `prototool` handles the install lazily/transparently. 

We're attempting to distribute a docker image with `protoc` and all of our required plugins baked in as a cohesive piece, and in order to get `prototool` to install `protoc`, we're doing this:

```dockerfile
ADD prototool.yaml /
# Required to make prototool install protoc for us
RUN prototool create temp.proto && prototool generate && rm temp.proto
```

It would be nice if there was a way to explicitly trigger `prototool` to install `protoc` in this case, e.g:
```dockerfile
ADD prototool.yaml /
RUN prototool install
```

Thanks!

**Comment by peter-edge** [comment]
We used to have this and were waiting to see if a user would request it after we removed it heh. @alexapb @amckinney thoughts?

**Comment by peter-edge** [comment]
@lwc I know this is super hacky, but this would accomplish what you need, including as a `RUN` directive for Docker:

```bash
rm -rf /tmp/prototool-bootstrap && \
  prototool config init /tmp/prototool-bootstrap && \
  echo 'syntax = "proto3";' > /tmp/prototool-bootstrap/tmp.proto && \
  prototool compile /tmp/prototool-bootstrap && \
  rm -rf /tmp/prototool-bootstrap
```

The better way to do this as part of a bash script would be to use `mktemp -d`, but for Darwin or Linux, the above should work. We could update the documentation to suggest this @amckinney @AlexAPB.

**Comment by peter-edge** [comment]
Also see https://github.com/uber/prototool/pull/226

**Comment by peter-edge** [comment]
Note if you want to re-add `prototool download`, you will want a way to specify the configuration to use. In the future, this might be stuff other than protoc, so a flag `--protoc-version` probably isn't the best. In this case, note that:

```bash
mkdir /tmp/prototool-bootstrap && \
  echo $'protoc:\n  version:' $PROTOC_VERSION > /tmp/prototool-bootstrap/prototool.yaml && \
  echo 'syntax = "proto3";' > /tmp/prototool-bootstrap/tmp.proto && \
  prototool compile /tmp/prototool-bootstrap
```

Will probably only be simplified to:

```bash
mkdir /tmp/prototool-bootstrap && \
  echo $'protoc:\n  version:' $PROTOC_VERSION > /tmp/prototool-bootstrap/prototool.yaml && \
  prototool download /tmp/prototool-bootstrap
```

With #224 however, this would just be:

```bash
prototool download --config-data '{"protoc":{"version":"3.6.1"}}'
```

Which would be nice.

**Comment by peter-edge** [comment]
Now that I've seen this pattern, one other thing I like about using `prototool compile` to do the pre-cache of `protoc` instead of having a customized command - it has the side-effect of making sure that the download worked properly, since `prototool compile` calls `protoc`. If anything, `download` should create a temporary file with just `syntax = "proto3";` in it, and call `compile` on it.

Of course the above documentation I think suffices as well, but I see the reasoning to add `download` back.

**Comment by AlexAPB** [comment]
This sounds reasonable to me, let's come back to this after #224 lands

**Comment by peter-edge** [comment]
I'd propose some combination of the following:

```
prototool cache update # does the download per the configuration file in the current directory and calls compile on a dummy file, similar to what is currently in the FAQ
prototool cache update . # the above is shorthand for this, similar to other commands
prototool cache update path/to/include # uses protoc.version in path/to/include/prototool.{json,yaml}, similar to other commands
prototool cache update --config-data '{"protoc:{"version":"3.6.1"}}' # same thing, but explicitly overrides any configuration file. i
prototool cache base-path # returns on mac ie Library/Caches/prototool/Darwin/x86_64, note this does not do the full path to the protoc directory as we could cache other things in the future
prototool cache protobuf-path # Library/Caches/prototool/Darwin/x86_64/protobuf/3.6.1
prototool cache delete # deletes on mac ie Library/Caches/Prototool
```

**Comment by peter-edge** [comment]
This is fixed in #257.
